.github/workflows/product_images.yml

```
@@ -1,18 +1,78 @@
 name: Product images
 
 on:
-  workflow_dispatch:
-    inputs:
-      product:
-        description: 'Product image to build.'     
-        required: true
-        default: 'zookeeper'
-      imageVersion:
-        description: 'Image version'  
+  push:
+    tags:
+      - '*'
+
 jobs:
+  reject_illegal_tags:
+    name: Check if tag adheres to format 'productx.y.z-stackablea.b.c'
+    runs-on: ubuntu-latest
+    outputs:
+      tag_name: ${{ steps.version_tag.outputs.tag_name }}
+    steps:
+      - name: Get version tag
+        id: version_tag
+        run: |
+          [[ ! "$GITHUB_REF" =~ refs/tags ]] && exit 255
+          TAG=${GITHUB_REF#refs/tags/}
+          echo "Found tag: [$TAG]"
+          echo "::set-output name=tag_name::$TAG"
+      - name: Check if tag matches expected format
+        env:
+          TAGNAME: ${{ steps.version_tag.outputs.tag_name }}
+        run: |
+          echo "Checking $TAGNAME"
+          [[ "$TAGNAME" =~ ^kafka|zookeeper|nifi|druid|opa|hbase|hdfs|trino|airflow|superset|spark-k8s|.+-stackable\d+\.\d+\.\d+ ]] && echo "${TAGNAME}" valid || exit 255
```

I have no idea, just noticed them missing in the list
```suggestion
          [[ "$TAGNAME" =~ ^commons|secret|kafka|zookeeper|nifi|druid|opa|hbase|hdfs|trino|airflow|superset|spark-k8s|.+-stackable\d+\.\d+\.\d+ ]] && echo "${TAGNAME}" valid || exit 255
```